PlatformRequirements.md

```diff
@@ -0,0 +1,144 @@
+    - It exposes a single 'prepare' method to provide a way for cordova-lib to apply a project's setting/www content to the platform.
+    - The PlatformApi class should be implemented by each platform that wants to use the new API flow. For those platforms, which don't provide their own PlatformApi, there will be a polyfill in cordova-lib.
+    - Platforms that implement their own PlatformApi instance should implement all prototype methods of this class to be fully compatible with cordova-lib.
+    - The PlatformApi instance should define the following field:
+        - platform : This is a String that defines a platform name.
+ 
+- Api.js exports static functions
+    - there is currently a requirement that the file be called Api.js (todo:change that)
+ 
+ 
+- Api.js exports static function `updatePlatform(destination, options, events);`
+    - PlatformApi.updatePlatform = function (cordovaProject, options) {};
+    - The `updatePlatform` method is equal to the bin/update script. It should update an already installed platform. It should accept a CordovaProject instance, that defines a project structure and configuration, that should be applied to the new platform, and an options object.
+    - cordovaProject: This is a CordovaProject instance that defines a project structure and configuration, that should be applied to the new platform. This argument is optional and if not defined, that platform is used as a standalone project and not as part of a Cordova project.
+    - options : This is an options object. The most common options are :
+        - options.customTemplate : This is a path to custom template, that should override the default one from platform.
+        - options.link : This is a flag that should indicate that the platform's sources will be linked to the installed platform instead of copying.
+     - The `updatePlatform` method must return a promise, which is either fulfilled with a PlatformApi instance or rejected with a CordovaError.
+ 
+- Api.js exports static function `createPlatform(destination, cfg, options, events);`
+    - PlatformApi.createPlatform = function(cordovaProject, options) {};
+    - The `createPlatform method` is equal to the bin/create script. It should install the platform to a specified directory and create a platform project. It should accept a CordovaProject instance, that defines a project structure and configuration, that should be applied to the new platform, and an options object.
+    - cordovaProject : This is a CordovaProject instance that defines a project structure and configuration, that should be applied to the new platform. This argument is optional and if not defined, that platform is used as a standalone project and not as part of a Cordova project.
+    - options : This is an options object. The most common options are :
+        - options.customTemplate : This is a path to custom template, that should override the default one from the platform.
```

Comment:
Voice should be 2nd person.  
Something like ... Templates allow app developers to create apps based on boilerplate application code ...
Not sure how deep you need to go as the audience for this guide is/are platform developers, so it should be more 'How to I respect the template parameter when it is given?' and not 'How do I use templates?'